test/unit/leafletData/leafletDataService.DirectiveControlsSpec.coffee

```diff
@@ -48,41 +63,59 @@ describe 'leafletData directiveControls', ->
                         fillColor: "blue"
                         color: 'white'
 
-            if preRunnerCb
-                preRunnerRet = preRunnerCb()
-
-            element = angular.element "<leaflet markers='markers' geojson='geojson'></leaflet>"
+            element = angular.element markup
             element = $compile(element)($rootScope)
 
             $rootScope.$digest()
-            leafletData.getDirectiveControls().then (controls) ->
-                if postRunnerCb
-                    postRunnerCb controls
 
-    describe 'markers controls', ->
-        beforeEach ->
-            @rootName = 'markers'
-        describe 'exists', ->
+            if Array.isArray id
+                return $q.all id.map (anId) ->
+                    leafletData.getDirectiveControls(anId)
+
+            leafletData.getDirectiveControls(id)
+
+    ['markers', 'geojson'].forEach (directiveName) ->
+        describe "#{directiveName} controls", ->
+            beforeEach ->
+                @rootName = directiveName
+                @controlsPromise = @testRunner()
+
             it 'root', ->
-                @testRunner (controls) =>
+                @controlsPromise
+                .then (controls) =>
                     expect(controls[@rootName]).toBeDefined()
             it 'create', ->
-                @testRunner (controls) =>
+                @controlsPromise
+                .then (controls) =>
                     expect(controls[@rootName].create).toBeDefined()
             it 'clean', ->
-                @testRunner (controls) =>
+                @controlsPromise
+                .then (controls) =>
                     expect(controls[@rootName].clean).toBeDefined()
 
-    describe 'geojson controls', ->
-        beforeEach ->
-            @rootName = 'geojson'
-        describe 'exists', ->
+
+        describe "#{directiveName} MANY controls", ->
+            beforeEach ->
```

Comment:
These tests were valuable especially after `obtainEffectiveMapId` was changed to throw on multiple maps and no mapId provided. Once the [main fix](https://github.com/angular-ui/ui-leaflet/pull/262/files#r66184840) was put in these test passed.